plugins/grpc-transport-auth/src/main/java/com/github/bsideup/liiklus/transport/grpc/config/GRPCTLSConfiguration.java

```diff
@@ -0,0 +1,109 @@
+    public void initialize(GenericApplicationContext applicationContext) {
+        var environment = applicationContext.getEnvironment();
+
+        var tlsProperties = PropertiesUtil.bind(environment, new GRPCTLSProperties());
+
+        if (tlsProperties.getKey() == null) {
+            return;
+        }
+
+        log.info("GRPC {}TLS ENABLED", tlsProperties.getTrustCert() != null ? "mutual " : "");
+
+        applicationContext.registerBean(
+                TLSGRPCTransportConfigurer.class,
+                () -> new TLSGRPCTransportConfigurer(tlsProperties)
+        );
+    }
+
+    @Value
+    static class TLSGRPCTransportConfigurer implements GRPCLiiklusTransportConfigurer {
+
+        GRPCTLSProperties properties;
+
+        @Override
+        public void apply(NettyServerBuilder builder) {
+            SslContext ctx =  TLSGRPCTransportConfigurer.createSSLContext(
+                    properties.getKey(),
+                    properties.getKeyPassword(),
+                    properties.getKeyCertChain(),
+                    properties.getTrustCert()
+            );
+
+            builder.sslContext(ctx);
+        }
+
+        /**
+         * Mostly copy of the https://github.com/grpc/grpc-java/tree/master/examples/example-tls
+         * and https://github.com/grpc/grpc-java/blob/master/SECURITY.md
+         * Currently all the values are paths, except the pwd. To make some room for extension,
+         * schema `file` is required, and the only supported for now.
+         *
+         * Refer to {@link io.netty.handler.ssl.SslContextBuilder#forServer(File keyCertChainFile, File keyFile, String keyPassword)}
+         * for more details.
+         *
+         * @param key          a PKCS#8 private key file in PEM format
+         * @param keyPassword  the password of the key or null if not protected
+         * @param keyCertChain an X.509 certificate chain file in PEM format
+         * @param trustCert    file should contain an X.509 certificate collection in PEM format
+         * @return ready-to-use ssl context.
+         */
+        @SneakyThrows
+        static SslContext createSSLContext(String key, String keyPassword, String keyCertChain, String trustCert) {
+            SslContextBuilder sslClientContextBuilder = SslContextBuilder.forServer(
+                    new File(keyCertChain.substring(FILE_PREFIX.length())),
+                    new File(key.substring(FILE_PREFIX.length())),
+                    keyPassword
+            );
+            if (trustCert != null) {
+                sslClientContextBuilder.trustManager(new File(trustCert.substring(FILE_PREFIX.length())));
+                sslClientContextBuilder.clientAuth(ClientAuth.REQUIRE);
+            }
+            return GrpcSslContexts.configure(sslClientContextBuilder).build();
+        }
+    }
+
+    @ConfigurationProperties("grpc.tls")
+    @Data
+    static class GRPCTLSProperties {
+
+        @Pattern(regexp = "^file://.*$")
+        String key;
```

Comment:
consider using Spring's resources here, AFAIK it will automatically resolve/map/verify